File: Core/Entities/Birthday/BirthdayPackage.cs
using System;
using System.Collections.Generic;
using Core.Entities.Birthday;

namespace Core.Entities
{
    public class BirthdayPackage : BaseEntity
    {
        public string PackageName { get; set; }
        public string Description { get; set; }
        public int NumberOfParticipants { get; set; }
        public decimal Price { get; set; }
        public decimal AdditionalBillingPerParticipant { get; set; }
        public int Duration { get; set; }
        public string Picture { get; set; }
        public decimal? DiscountedPrice { get; set; }
        public bool? HasDiscountsApplied { get; set; }


        public ICollection<BirthdayPackageService> BirthdayPackageServices { get; set; }
        public ICollection<BirthdayPackageDiscount> BirthdayPackageDiscounts { get; set; }
        public ICollection<Birthday1> Birthdays { get; set; }

    }
}

File: client/src/app/shared/models/birthdays/birthdaypackage.ts
import { Discount } from '../discount';
import { ServiceIncluded } from './serviceincluded';

export interface BirthdayPackage {
    id: number;
    packageName: string;
    description: string;
    numberOfParticipants: number;
    price: number;
    additionalBillingPerParticipant: number;
    duration: number;
    picture: string;
    discountSum: number;
    discountedPrice?: number;
    hasDiscountsApplied?: boolean;
    servicesIncluded: ServiceIncluded[];
}

export interface BirthdayPackageCreateEdit {
    id: number;
    packageName: string;
    description: string;
    numberOfParticipants: number;
    price: number;
    additionalBillingPerParticipant: number;
    duration: number;
    picture: File;
    discountsIds: number[];
    servicesIds: number[];
}

export class BirthdayPackageCreateEditClass {
    id: number;
    packageName: string;
    description: string;
    numberOfParticipants: number;
    price: number;
    additionalBillingPerParticipant: number;
    duration: number;
    picture: File;
    discountsIds: number[];
    servicesIds: number[];
}

export interface BirthdayPackagePutGet {
    BirthdayPackage: BirthdayPackage;
    selectedServices: ServiceIncluded[];
    nonSelectedServices: ServiceIncluded[];
    selectedDiscounts: Discount[];
    nonSelectedDiscounts: Discount[];
}










